src/web/js/check-ui.js

```diff
@@ -24,246 +24,426 @@
+          (isPassing ? "hsl(88, 50%, 76%)" : "hsl(45, 100%, 85%)"));
+        let name   = nameHandle.anchor;
+        let handle = nameHandle.handle;
+        let status = document.createTextNode(isPassing ? ": Passed" : ": Failed");
+        header.appendChild(name);
+        header.appendChild(status);
+        return {header : header, handle : handle};
+      }
+      
+      let lastHighlighted = undefined;
+      
+      var FailingTestSkeleton = function () {
+        function FailingTestSkeleton(test, testNumber) {
+          var container = document.createElement("div");
+          var headerHandle = makeTestHeader(testNumber, get(test, "loc"), false);
+          var header = headerHandle.header;
+          var handle = headerHandle.handle;
+          var tombstone = document.createElement("div");
+          container.classList.add("check-block-test");
+          container.classList.add("failing-test");
+          tombstone.classList.add("test-reason");
+          container.appendChild(header);
+          container.appendChild(tombstone);
+          var thisTest = this;
+          var source = get(get(test, "loc"), "source");
+          if (documents.has(source)) {
+            var doc = documents.get(source);
+            var editor   = doc.getEditor();
+            if (editor !== undefined) {
+              var anchor = makeGutterMarker(handle);
+              anchor.classList.add("failed-test-marker");
+              anchor.addEventListener("click", function (e) {
+                thisTest.block.showTest(thisTest);
+              });
+            }
+          }
+          
+          if(runtime.hasField(test, "actual-exn")) {
+            this.maybeStackLoc = outputUI.makeMaybeStackLoc(
+              runtime, documents, srcloc, get(test, "actual-exn").val.pyretStack);
+          } else {
+            this.maybeStackLoc = noFramesMaybeStackLoc;
+          }
+          this.renderable = test;
+          this.container = container;
+          this.tombstone = tombstone;
+        }
 
+        FailingTestSkeleton.prototype.highlight = function highlight() {
+          outputUI.clearEffects();
+          if (this.rendering) {
+            this.rendering.addClass("highlights-active");
+            this.rendering.trigger("toggleHighlight");
+          }
+          lastHighlighted = this;
+          lastHighlighted.container.classList.add("highlights-active");
+          lastHighlighted.tombstone.classList.add("highlights-active");
+        };
 
-    // NOTE: MUST BE CALLED WHILE RUNNING ON runtime's STACK
-    function drawCheckResults(container, editors, runtime, checkResults, contextFactory) {
-      var ffi = runtime.ffi;
-      var cases = ffi.cases;
-      var get = runtime.getField;
+        FailingTestSkeleton.prototype.refresh = function refresh() {
+          var snippets = this.tombstone.querySelectorAll(".CodeMirror");
+          for (var i = 0; i < snippets.length; i++) {
+            snippets[i].CodeMirror.refresh();
```

Comment:
If there are a large number (i.e. more than ~30) of tests in this skeleton, this loop will cause noticeable jankiness as the browser freezes for a split-second.  Ditto elsewhere.  The example we were using to stress this was a loop `for each(i in range(0, 250)): num-modulo(i, 2) is 1 end`, which easily triggered a lot of failing tests all in one test block (and also all in one line of code).  How can this code avoid the jank?  (Same comment applies elsewhere, for any loop over all snippets' CodeMirrors.)